### ISSUE TITLE
Have a FactoryHashmap to allow changing factory elements based on keys.

### ISSUE THREAD

**Issue by 13hannes11** [Relm4/Relm4#181]
This could be an alternative to `FactoryVecDeque` for use cases where a Hashmap is more appropriate.

**Comment by edfloreshz** [comment]
I will try to tackle this issue, not sure if I'll accomplish anything so, if other's want to do it, feel free.

**Comment by AaronErhardt** [comment]
@edfloreshz thanks! I think it makes sense to use the next branch because its factory implementation a lot easier to understand and should translate well to HashMaps: https://github.com/Relm4/Relm4/blob/next/relm4/src/factory/mod.rs

I happy to help if you have any questions.

**Comment by edfloreshz** [comment]
Thanks @AaronErhardt, that's helpful. I'll take a look at it and let you know if I have questions.

**Comment by songww** [comment]
The types in many places in the code are hardcoded as DynamicIndex. but this type is not suitable to be used as a key in a HashMap.

**Comment by AaronErhardt** [comment]
That's true. A HashMap can't really use DynamicIndex because elements cannot be reordered in a traditional sense and are already identified by their key. Adding a HashMap will likely require adjustments to FactoryComponent.

**Comment by 13hannes11** [comment]
@AaronErhardt this seems to have been implemented as I was using this on the latest version I will thus close this issue.
